src/test/java/eu/yals/test/ui/VaadinTest.java

```diff
@@ -0,0 +1,197 @@
+import org.junit.Rule;
+import org.junit.rules.TestRule;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Cookie;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.ProtocolHandshake;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Service class for all Vaadin aka UI Tests.
+ *
+ * @since 2.7
+ */
+@Slf4j
+public abstract class VaadinTest extends ParallelTest {
+    static {
+        Logger sauceLabsLog = Logger.getLogger(SauceLabsIntegration.class.getName());
+        Logger selenideRemoteLog = Logger.getLogger(ProtocolHandshake.class.getName());
+
+        sauceLabsLog.setLevel(Level.SEVERE);
+        selenideRemoteLog.setLevel(Level.WARNING);
+    }
+
+    private static final int SERVER_PORT =
+            Integer.parseInt(
+                    System.getProperty(TestApp.Properties.SERVER_PORT, TestApp.Defaults.SERVER_PORT));
+    private static final String LOCAL_URL =
+            String.format("http://host.testcontainers.internal:%d", SERVER_PORT);
+    private static final String REPORT_DIRECTORY =
+            System.getProperty(TestApp.Properties.REPORT_DIR, TestApp.Defaults.REPORT_DIR);
+    protected static final String BASE_URL =
+            System.getProperty(TestApp.Properties.TEST_URL, LOCAL_URL);
+
+    private static final String BUILD_NAME =
+            System.getProperty(TestApp.Properties.BUILD_NAME, TestApp.Defaults.BUILD_NAME);
+
+    private static String testName;
+
+    @Rule
+    public final TestRule watchman =
+            new TestWatcher() {
+                @Override
+                protected void starting(final Description description) {
+                    super.starting(description);
+                    testName = setTestNameFromTestDescription(description);
+                    log.info(String.format("Starting build '%s'. Test: '%s", BUILD_NAME, testName));
+                }
+
+                @Override
+                protected void succeeded(final Description description) {
+                    super.succeeded(description);
+                    Cookie cookie = new Cookie("zaleniumTestPassed", "true");
+                    getDriver().manage().addCookie(cookie);
+                }
+
+                @Override
+                protected void failed(final Throwable e, final Description description) {
+                    Cookie cookie = new Cookie("zaleniumTestPassed", "false");
+                    getDriver().manage().addCookie(cookie);
+                }
+            };
+
+    /**
+     * Performs actions needed for running tests.
+     *
+     * @throws Exception if error occurred
+     */
+    @Before
+    public void setup() throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [You may have misspelled a JUnit framework method (setUp or tearDown)](https://app.codacy.com/gh/yadevee/yals/pullRequest?prid=5487211)